File: src/main/java/com/terpomo/wavy/util/Point.java
package com.terpomo.wavy.util;

import com.terpomo.wavy.marshal.IMarshallable;
import com.terpomo.wavy.marshal.MarshalAttr;
import com.terpomo.wavy.marshal.MarshallingKeys;

public class Point implements IMarshallable {
    private float x, y;

    public Point() {
    }

    public Point(float x, float y) {
        this.x = x;
        this.y = y;
    }

    @MarshalAttr(attrName= MarshallingKeys.KEY_X)
    public void setX(float x) {
        this.x = x;
    }

    @MarshalAttr(attrName= MarshallingKeys.KEY_Y)
    public void setY(float y) {
        this.y = y;
    }

    @MarshalAttr(attrName= MarshallingKeys.KEY_X)
    public float getX() {
        return x;
    }

    @MarshalAttr(attrName= MarshallingKeys.KEY_Y)
    public float getY() {
        return y;
    }
}

File: msvc-shared-library/math/point.cpp
#include "point.h"

Point::Point() : Point::Point(0.0f, 0.0f) {};

Point::Point(float x, float y) {
	this->x = x;
	this->y = y;
}

void Point::setX(float x) {
	this->x = x;
}

void Point::setY(float y) {
	this->y = y;
}

float Point::getX() {
	return this->x;
}

float Point::getY() {
	return this->y;
}

Point Point::zero() {
	return Point(0.0f, 0.0f);
}